### DISCUSSION TITLE
[qunit] Add type for QUnit.test.each()

### DISCUSSION THREAD

**Discussion by Krinkle** [DefinitelyTyped/DefinitelyTyped#59733] [Issues with a @types package]
https://api.qunitjs.com/QUnit/test.each/

For example:

```js
QUnit.test.each('isEven()', [2, 4, 6], (assert, data) => {
  assert.true(isEven(data), `${data} is even`);
});

QUnit.test.each('isEven()', {
  caseEven: [2, true],
  caseNotEven: [3, false]
}, (assert, data) => {
  assert.strictEqual(isEven(data[0]), data[1]);
});

QUnit.test.each('isEven()', {
  caseEven: [2, true],
  caseNotEven: [3, false]
}, (assert, [value, expected]) => {
  assert.strictEqual(isEven(value), expected);
});
```

In a nut shell:
* Much like `QUnit.test`.
* Requires a second parameter that's an array or object.
* The callback is repeatedly called for each of the values of that array or object as second parameter after `assert`; akin to `Array#forEach` in terms of the iteration and pass-through behaviour.

[a comment by an automated account is not shown]

**Comment by chriskrycho** [comment]
Hey @Krinkle, I just saw you opened the PR at #60677—sorry we all missed that! Summer is rough with vacations etc. If you want to get this across the line, please feel free to reopen that PR and ping me on either the Ember Discord or the TypeScript Discord!

**Comment by sukima** [comment]
Hey everyone! it has been well over two years. What is blocking `QUnit.test.each()`? It is **very** unfortunate many developers are unable to use this feature in qunit.
